pkg/nfd-master/updater-pool.go

```diff
@@ -158,18 +154,14 @@ func (u *updaterPool) start(parallelism int) {
 			go u.runNodeFeatureGroupUpdater(u.nfgQueue)
 		}
 	}
+	u.started = true
 }
 
 func (u *updaterPool) stop() {
 	u.Lock()
 	defer u.Unlock()
 
-	if u.queue == nil || u.queue.ShuttingDown() {
-		klog.InfoS("the NFD master updater pool is not running.")
-		return
-	}
-
-	if u.nfgQueue == nil || u.nfgQueue.ShuttingDown() {
+	if !u.started {
```

Comment:
why don't we use the u.running method?

Reply:
As that's also behind a lock

Reply:
perfect thanks, didn't see the first lock 😄